master/member.go

```diff
@@ -28,34 +28,28 @@ func (m *Member) Unmarshal(data []byte) error {
 	return json.Unmarshal(data, m)
 }
 
-// nolint:unused
 func (s *Server) updateServerMasterMembers(ctx context.Context) error {
-	leader, exists := s.checkLeader()
 	resp, err := s.etcdClient.MemberList(ctx)
 	if err != nil {
 		return err
 	}
+
 	members := make([]*Member, 0, len(resp.Members))
-	currLeader, exist := s.checkLeader()
+	leader, exists := s.checkLeader()
+	etcdLeaderID := s.etcd.Server.Lead()
 	for _, m := range resp.Members {
 		isServLeader := exists && m.Name == leader.Name
-		if isServLeader && (!exist || currLeader.Name != m.Name) {
-			s.leader.Store(&Member{
-				Name:         m.Name,
-				IsServLeader: true,
-				IsEtcdLeader: true,
-				Addrs:        m.ClientURLs,
-			})
-			s.createLeaderClient(ctx, m.ClientURLs)
-		}
+		isEtcdLeader := m.ID == etcdLeaderID
```

Comment:
what's the relationship between the server leader and. etcd leader ?

Reply:
etcd leader is elected first, which is based the raft election in embed etcd.
server leader is based on etcd leader, but it could have short latency, and we can observe 
- etcd leader, server leader in the same node
- etcd leader exists, server leader doesn't exist for a short time.